src/System.Net.Mail/tests/Functional/SmtpClientTest.cs

```diff
@@ -363,5 +363,73 @@ public async Task TestCredentialsCopyInAsyncContext()
                 server.Stop();
             }
         }
+
+
+        [Theory]
+        [InlineData(false, false, false)]
+        [InlineData(false, false, true)] // Received text.
+        [InlineData(false, true, false)]
+        [InlineData(false, true, true)]
+        [InlineData(true, false, false)]
+        [InlineData(true, false, true)] // Received text.
+        [InlineData(true, true, false)]
+        [InlineData(true, true, true)]
+        public void SendMail_DeliveryFormatSevenBit_SubjectEncoded(bool useAsyncSend, bool useSevenBit, bool useSmtpUTF8)
```

Comment:
You should split this out into multiple tests. That way, if one test fails it is more immediately obvious what went wrong. I would suggest you at least split it into seven bit and utf-8 versions, but don't feel strongly about whether or not you should split each one again for sync vs async.

Reply:
Perhaps it's a problem with method names. There's almost no difference between `SevenBit` and `International` test code. I changed the method name to `SendMail_DeliveryFormat_SubjectEncoded`, which should be more suitable for this test.

The last test here will go wrong, and I've put him in the last place. `[InlineData(true, true, true)]`. Other tests will not be problematic.